src/development/platform-integration/android/restore-state-android.md

```diff
@@ -0,0 +1,185 @@
+state of the app before it was killed,
+you've lost the state and the app starts from scratch.
+The user has lost the continuity they expect,
+which is clearly not ideal.
+(Imagine filling out a lengthy form and being interrupted
+by a phone call _before_ clicking **Submit**.)
+
+So, how can you restore the state of the app so that
+it looks like it did before it was sent to the
+background?
+
+Flutter has a solution for this with the
+[`RestorationManager`][] (and related classes)
+in the [services][] library.
+With the `RestorationManager`, the Flutter framework
+provides the state data to the engine _as the state
+changes_, so that the app is ready when the OS signals
+that it's about to kill the app, giving the app only
+moments to prepare.
+
+{{site.alert.secondary}}
+  **Instance state vs long-lived state**
+  When should you use the `RestorationManager` and
+  when should you save state to long term storage?
+  _Instance state_
+  (also called _short-term_ or _ephemeral_ state),
+  includes unsubmitted form field values, the currently
+  selected tab, and so on. On Android, this is
+  limited to 1 MB and, if the app exceeds this,
+  it crashes with a `TransactionTooLargeException`
+  error in the native code.
+{{site.alert.end}}
+
+[state]: {{site.url}}/development/data-and-backend/state-mgmt/ephemeral-vs-app
+
+## Overview
+
+You can enable state restoration with just a couple tasks:
+
+1. Define a `restorationId` or a `restorationScopeId`
+   for all widgets that support it,
+   such as [`TextField`][] and [`ScrollView`][].
+   This automatically enables built-in state restoration
+   for those widgets.
+
+2. For custom widgets,
+   you must decide what state you want to restore
+   and hold that state in a [`RestorableProperty`][].
+   (The Flutter API provides various subclasses for
+   different data types.)
+   Define those `RestorableProperty` widgets 
+   in a `State` class that uses the [`RestorationMixin`][].
+   Register those widgets with the mixin in a
+   `restoreState` method.
+
+Other considerations:
+
+* Providing a `restorationId` to
+  `MaterialApp`, `CupertinoApp`, or `WidgetsApp`
+  automatically enables state restoration by
+  injecting a `RootRestorationScope`.
+  If you need to restore state _above_ the app class,
+  inject a `RootRestorationScope` manually.
+
+* **The difference between a `restorationId` and
+  a `restorationScopeId`:** Widgets that take a
+  `restorationScopeID` create a new `restorationScope`
+  (a new `RestorationBucket`) into which all children
+  store their state. A `restorationId` means the widget
+  (and its children) store the data in the surrounding bucket.
+  (QUESTION: Axe this?)
+
+* You can only restore state with the `RestorationMixin`
+  that is stored in `RestorableProperty` (and its subclasses),
+  for example, [`restorablePush`][].
+  To restore the state in those properties,
+  you have to register them in `restoreState`.
+  (QUESTION: Axe this?)
```

Comment:
I would axe this bullet point. It kinda just repeats information given previously.